portfolio/src/main/java/com/google/sps/data/Comment.java

```diff
@@ -34,6 +87,8 @@
   public Comment(String author, String commentBody) {
     this.author = author;
     this.commentBody = commentBody;
+    this.id = -1;
+    this.timePosted = -1;
```

Comment:
```suggestion
    this.timePosted = System.currentTimeMillis();
```
Would this make more sense?

Reply:
No, although that idea maybe brings into question if this class is being handled the best that it could be (or maybe even just being commented as best as it could be). ID and time posted are tied to the database entry, so when the comment object is referencing data that has yet to be posted to a database those are expected to be their default values of -1. Is this bad style or should I add more clarifying comments?

Reply:
Yeah. I brought this up because I was looking at the code in DataServlet, lines 106-109. When you have `comment.fillEntity` and then a few lines later `return new Comment(commentEntity)` it's a little unclear to me what the code is really doing. I have to go back here and read all the relevant functions and constructors.

I can see how we're trying to reduce code by writing these helper functions, but maybe it would help readability just to be a little more explicit? I guess all I'm asking is, since we have `comment` already with all the fields set except the datastore ID, why not just grab the new ID and have a setId() function, instead of having this extra constructor?

Reply:
Yeah taking a look again that function you bring up looks a little spaghetti-like. I boiled it down to the two ultimate use-cases for comments: making a new comment from a post request or getting an existing comment from the database. So I left only two constructors. One that only sets the non-DB params and gets the current time for the posted time. The other that constructs the comment based on a DB entity. Looks much better now, thanks!